### PR TITLE
Made changes to documentation regarding deprecation and usage of set_ptr

### PR DESCRIPTION
**Pull request by mbag**
Since set_ptr arguemnt to resource record is creating PTR record automatically, but not deleting it. I have made changes to documentation to warn users about this feature. I also added deprecationwarning for this attribute, since PowerDNS API notes that set_ptr will be deprecated in version v4.3.0

### CODE REVIEW

**Comment by mbag** [comment]
@splashx can you review these docs changes?

File: website/docs/r/record.html.markdown [hunk 1]
```diff
@@ -12,8 +12,9 @@ Provides a PowerDNS record resource.
 
 ## Example Usage
 
-Note that PowerDNS internally lowercases certain records (e.g. CNAME and AAAA), which can lead to resources being marked for a change in every singe plan.
+Note that PowerDNS internally lowercases certain records (e.g. CNAME and AAAA), which can lead to resources being marked for a change in every single plan/apply.
```
Comment (splashx) [inline website/docs/r/record.html.markdown]: I would add a hint of _future-proof_ writing on this part:
> that PowerDNS **may** internally ...

> which **may** lead to ...

File: website/docs/r/record.html.markdown [hunk 2]
```diff
@@ -40,9 +42,17 @@ resource "powerdns_record" "foobar" {
...
+### Automatically set PTR record for A/AAAA records
+
+!> **Deprecation warning:** _set_ptr_ feature is set to be deprecated in PowerDNS v4.3.0
+
+PowerDNS API v4.2.0 offers a feature to automatically create corresponding PTR record for the A/AAAA record.
+Existing PTR records with the same name are replaced. If no matching reverse zone is found, resource creation will fail.
 You can use `powerdns_zone` resource to create the reverse zone.
+
+
+!> **Warning:** Using _set_ptr:true_  will not automatically remove the PTR record when A/AAAA record is deleted. You should create PTR zone using `powerdns_zone` and manage PTR records using `powerdns_record`, rather than using _set_ptr_. With upcoming _set_ptr_ deprecation, this will become default way of maintaining PTR records.
```
Comment (splashx) [inline website/docs/r/record.html.markdown]: > this will become default way of maintaining PTR records.

I would rephrase that to: _this will be the only way of maintaining PTR records **via this provider**_.

File: website/docs/r/record.html.markdown [hunk 3]
```diff
@@ -78,6 +88,7 @@ The following arguments are supported:
 * `type` - (Required) The record type.
 * `ttl` - (Required) The TTL of the record.
 * `records` - (Required) A string list of records.
+* `set_ptr` - (Optional) A boolean (true/false), determining whether API server should automatically create PTR record in the matching reverse zone. Existing PTR records are replaced. If no matching reverse zone, an error is thrown.
```
Comment (splashx) [inline website/docs/r/record.html.markdown]: I would add here already a text maybe at the end: _Deprecated in PowerDNS 4.3.0_.

Comment (mbag) [reply website/docs/r/record.html.markdown]: done

File: website/docs/r/record.html.markdown (hunk 1 above)
Comment (mbag) [reply website/docs/r/record.html.markdown]: done

Comment (mbag) [reply website/docs/r/record.html.markdown]: done

File: website/docs/r/record.html.markdown (hunk 2 above)
Comment (mbag) [reply website/docs/r/record.html.markdown]: done
